helper/winsvc/mock_windows_service.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package winsvc
```

Comment:
I can appreciate providing a mock here, but it's unusual to define a mock outside the test package that uses it, unless it's imported by multiple other packages that all use it the same.